src/search.js

```diff
@@ -0,0 +1,125 @@
+var HMS = require('hindley-milner-search');
+
+var state = {
+  // [Node], the dom elements we will be showing/hiding when user searches
+  els: null,
+  // Node, container for results
+  container: null,
+  // Node, container for notification if there are no matches
+  containerNo: null,
+  // [Signature], the signatures as compiled by HMS
+  index: [],
+  // [Signature], most recent matches
+  matches: [],
+  // String, most recent user input
+  prevSearch: 'placeholder',
+};
+
+function map(f, x) {
+  return Array.prototype.map.call(x, f);
+}
+
+// Node -> Node
```

Comment:
I'd like to be able to run `ack '^ *//' src/search.js` to view all the type signatures. Please update this signature (and others) to include the function name:

```javascript
//  getGrandparent :: Node -> Node
```